File: content/LLD/Reference-Code/1. OOPS Recap/1. Revision/inheritance.py
class Animal:
    def __init__(self, name: str, age: int):
        print("Animal INIT")
        self.name = name
        self.age = age

    def eat(self):
        print("I am eating")

    def sleep(self):
        print("I am sleeping")

    def move(self):
        print("I am moving")


class Dog(Animal):
    def __init__(self, name: str, age: int, breed: str):
        super().__init__(name, age)
        self.breed = breed

    def bark(self):
        print("I am barking")

    def display(self):
        print(f"Name is {self.name} and age is {self.age}")

    def move(self):
        print("I am running on 4 legs")


dog = Dog("Cheery", 5, "Indie")
dog.move()

File: content/LLD/Reference-Code/1. OOPS Recap/1. Revision/Inheritance.java
class Animal {
    protected String name;
    protected int age;

    public Animal(String name, int age) {
        System.out.println("Animal INIT");
        this.name = name;
        this.age = age;
    }

    public void eat() {
        System.out.println("I am eating");
    }

    public void sleep() {
        System.out.println("I am sleeping");
    }

    public void move() {
        System.out.println("I am moving");
    }
}

class Dog extends Animal {
    private String breed;

    public Dog(String name, int age, String breed) {
        super(name, age);
        this.breed = breed;
    }

    public void bark() {
        System.out.println("I am barking");
    }

    public void display() {
        System.out.println("Name is " + this.name + " and age is " + this.age);
    }

    @Override
    public void move() {
        System.out.println("I am running on 4 legs");
    }
}

public class Inheritance {
    public static void main(String[] args) {
        Dog dog = new Dog("Cheery", 5, "Indie");
        dog.move();
    }
}
